wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +456,143 @@ def get_device_unique_id(device):
 
     def disconnect(self):
         self.logger.info("LenovoSystem disconnected")
+
+    def __getattr__(self, attr):
+        """
+        Tries to delegate a not found method to objects inside self._delegate
+        :raises AttributeError if ``attr`` is not found in any objects this should delegate
+        """
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                return getattr(delegate_to, attr)
+        raise AttributeError(attr)
+
+
+class LenovoSwitches(object):
```

Comment:
Please consider extending this as a `wrapanapi/entities/base/Entity` class, since it is an object on the LenovoSystem provider.  This will provide access to the `Entity.system` attribute, linking back to the `LenovoSystem` object tied to this switch item.

This would then be a singluar LenovoSwitch object, and `LenovoSystem` would have class methods to deal with multiple switches - returning a list of switch objects, finding a switch object, etc.

Such a design would match the intention of wrapanapi 3.0